### ISSUE TITLE
Snakemake using different values of "threads" for console output and cluster submission

### ISSUE THREAD

**Issue by corneliusroemer** [snakemake/snakemake#1091]
**Describe the bug**
When I submit a job to the cluster, the shell command that actually gets executed on the cluster differs from what Snakemake tells me it is running in its terminal output. What differs in particular is the value of `threads`.

It seems that for printing the shell command in the log/terminal, Snakemake assumes it's running locally. So `threads` is set to the number it would be if this rule was local, even if it's not. For example if `cores: 4` is set in the profile, the number of threads printed in the terminal will never exceed 4, even if the command submitted to the cluster will actually use a value of threads that can be higher.

**Logs**
This is what I get told is being run:
![Screenshot 2021-07-14 at 15 06 22](https://user-images.githubusercontent.com/25161793/125627384-2697814d-a2ac-4b07-af1a-4db8c67ec99f.png)

And this is what is actually running on the worker node on the cluster. In the code, `jobs` is set as follows: `--jobs {threads}`. `{threads}` should always be 8, but it's displayed as 4 in the terminal (see above).

![Screenshot 2021-07-14 at 15 06 01](https://user-images.githubusercontent.com/25161793/125627399-49b33421-5022-4ea0-9ccc-207ad4bd1cea.png)
![Screenshot 2021-07-14 at 15 05 51](https://user-images.githubusercontent.com/25161793/125627404-589f8791-2559-44ea-8db0-b3a855de4657.png)

**Additional context**
Snakemake version: 6.5.2

**Comment by johanneskoester** [comment]
Good catch! This sounds indeed like a bug. If you could provide a testcase, I can try to fix it ASAP.
